### PR TITLE
Use RuboCop's code for generating documentation

### PR DESCRIPTION
**Pull request by bquorning**
Rubocop now has a `CopsDocumentationGenerator` class that can generate consistent documentation across all projects.

With this PR, we will be updating the `docs` folder and no longer updating the `manual` folder. Should we delete it? Which documentation is currently published, and will it work after merging this PR?

---

Before submitting the PR make sure the following are checked:

* [x] Feature branch is up-to-date with `master` (if not - rebase it).
* [x] Squashed related commits together.
* [-] Added tests.
* [x] Updated documentation.
* [-] Added an entry to the `CHANGELOG.md` if the new code introduces user-observable changes.
* [x] The build (`bundle exec rake`) passes (be sure to run this locally, since it may produce updated documentation that you will need to commit).

### CODE REVIEW

**Comment by bquorning** [comment]
As per #972, I think we should now be publishing to https://docs.rubocop.org/rubocop-rspec/ instead of https://rubocop.readthedocs.io/projects/rspec/en/latest/

**Comment by Darhazer** [comment]
> As per #972, I think we should now be publishing to https://docs.rubocop.org/rubocop-rspec/ instead of https://rubocop.readthedocs.io/projects/rspec/en/latest/

Have no idea what to configure there though

File: rubocop-rspec.gemspec [hunk 1]
```diff
@@ -33,7 +33,7 @@ Gem::Specification.new do |spec|
 
   spec.metadata = {
     'changelog_uri' => 'https://github.com/rubocop-hq/rubocop-rspec/blob/master/CHANGELOG.md',
-    'documentation_uri' => 'https://rubocop-rspec.readthedocs.io/'
+    'documentation_uri' => 'https://docs.rubocop.org/rubocop-rspec/'
   }
 
   spec.add_runtime_dependency 'rubocop', '~> 0.87'
```
Comment (pirj) [inline rubocop-rspec.gemspec:39]: To generate docs, do we depend on 0.89 In development?

File: Rakefile [hunk 2]
```diff
@@ -53,11 +53,11 @@ end
 desc 'Confirm documentation is up to date'
 task confirm_documentation: :generate_cops_documentation do
   _, _, _, process =
-    Open3.popen3('git diff --exit-code manual/')
+    Open3.popen3('git diff --exit-code docs/')
 
   unless process.value.success?
     raise 'Please run `rake generate_cops_documentation` ' \
-          'and add manual/ to the commit.'
+          'and add docs/ to the commit.'
```
Comment (pirj) [inline Rakefile:60]: Wondering if this can/should be rephrased to "stage your doc/ changes".

File: lib/rubocop/cop/rspec/base.rb [hunk 3]
```diff
@@ -29,6 +29,8 @@ class Base < ::RuboCop::Cop::Base
                                .map(&Regexp.public_method(:new))
         )
 
+        exclude_from_registry
```
Comment (pirj) [inline lib/rubocop/cop/rspec/base.rb:32]: 👍

File: rubocop-rspec.gemspec (hunk 1 above)
Comment (bquorning) [reply rubocop-rspec.gemspec:39]: Yes

File: Rakefile (hunk 2 above)
Comment (bquorning) [reply Rakefile:60]: I am concerned whether everybody would understand the verb “stage”. And after staging the changes, they _do_ need to also be committed :-)

**Comment by bbatsov** [comment]
> With this PR, we will be updating the docs folder and no longer updating the manual folder. Should we delete it? Which documentation is currently published, and will it work after merging this PR?

Yeah, you can delete the manual folder. Afterwards you can setup redirects in the RTD admin console if you want. 

> Which documentation is currently published, and will it work after merging this PR?

Nothing will happen automatically after merging this PR, as documentation deployments for the new system are currently triggered manually (typically after we cut some new release). I saw that @koic already explained the process in another ticket, and it's already in the docs somewhere - https://docs.rubocop.org/rubocop/0.89/contributing.html#working-on-the-docs Probably we can add there some more instructions down the road.

**Comment by bquorning** [comment]
Force-pushed with updated CHANGELOG.

**Pull request merged**